examples/jsm/utils/CameraUtils.js

```diff
@@ -64,10 +66,33 @@ function frameCorners( camera, bottomLeftCorner, bottomRightCorner, topLeftCorne
 		camera.fov =
 			MathUtils.RAD2DEG / Math.min( 1.0, camera.aspect ) *
 			Math.atan( ( _vec.copy( pb ).sub( pa ).length() +
-							( _vec.copy( pc ).sub( pa ).length() ) ) / _va.length() );
+				( _vec.copy( pc ).sub( pa ).length() ) ) / _va.length() );
 
 	}
 
 }
 
-export { frameCorners };
+/** Calculate a PerspectiveCamera frustum's corners at a given distance from the camera.
+ * Returns an object with topLeft, topRight, bottomRight, and bottomLeft properties. Each property is a Vector3.
+ * Should work on all PerspectiveCameras where the frustum is rectangular in shape.
+ * Including asymmetric rectangular frustums like those used by webXR cameras.
+ * @param {number} distance - Distance from the camera
+ * @param {camera} camera - The PerspectiveCamera to calculate the frustum corners from */
+function getFrustumCorners( camera, distance ) {
+
+	camera.getBounds( distance, _minTarget, _maxTarget );
+
+	camera.updateMatrixWorld( true, false );
+
+	// .set() -> Calculates the corner position
+	// .applyMatrix4() -> Accounts for camera position/rotation/scale
+	return {
+		topLeft: _vec.set( _minTarget.x, _maxTarget.y, - distance ).applyMatrix4( camera.matrixWorld ).clone(),
```

Comment:
Please do not return an object or call `clone()`. The API should accept `topLeft` as an arg, instead.

Reply:
We then need to add `topRight`, `bottomRight` and `bottomLeft` as well which makes the signature a bit messy.

Maybe a signature could be `getFrustumCorners( camera, distance, target )` where `target` is an object that holds `topLeft`, `topRight`, `bottomRight` and `bottomLeft` as properties.

Reply:
Method removed for now. We can continue in a future PR/Issue 👌